src/nvim/eval.c

```diff
@@ -17685,6 +17685,15 @@ static void f_writefile(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     EMSG2(_(e_listarg), "writefile()");
     return;
   }
+  const list_T *const list = argvars[0].vval.v_list;
+  if (list == NULL) {
+    return;
+  }
+  TV_LIST_ITER_CONST(list, li, {
```

Comment:
TV_LIST_ITER_CONST is perfectly fine with a NULL list. Won’t execute a list body even once obviously, but still fine.

Reply:
So you should be able to just remove that `if()` block and everything would work as intended.